90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -1012,39 +1123,22 @@ load_be_cmdline() {
   if [ -e "${BASE}/noresume" ]; then
     zdebug "${BASE}/noresume set, processing ${zfsbe_args}"
     # Must replace resume= arguments and append a noresume
-    zfsbe_args="$( awk <<< "${zfsbe_args}" '
-      BEGIN {
-        quot = 0;
-        supp = 0;
-        ORS = " ";
-      }
-
-      {
-        for (i=1; i <= NF; i++) {
-          if ( quot == 0 ) {
-            # If unquoted, determine if output should be suppressed
-            if ( $(i) ~ /^resume=/ ) {
-              # Argument starts with "resume=", suppress
-              supp = 1;
-            } else {
-              # Nothing else is suppressed
-              supp = 0;
-            }
-          }
+    zfsbe_args="$( suppress_kcl_arg resume "${zfsbe_args}" ) noresume"
+  fi
 
-          # If output is not suppressed, print the field
-          if ( supp == 0 && length($(i)) > 0 ) {
-            print $(i);
-          }
+  # shellcheck disable=SC2154
+  if [ "${zbm_set_hostid}" -eq 1 ]; then
+    # Try to assume a matched hostid if none is defined
+    if [ -z "${spl_hostid}" ] && [ -r "${BASE}/spl_hostid" ]; then
```

Comment:
It would be nice to grab this from /etc/hostid instead of a byproduct of host matching, but reading the binary /etc/hostid requires an external tool. If we install `od`, we can grab it with `od -t x -An /etc/hostid`. However, reading it from this byproduct paves the way for per-BE hostids if we want them.

Reply:
`od` is part of coreutils, I don't think it's unreasonable to require it - especially if we want to do the right thing on MUSL systems.